modules/tribes/tests/client/components/TribesPage.component.test.js

```diff
@@ -0,0 +1,177 @@
+
+api.read.mockImplementation(async () => dummyTribes);
+api.join.mockImplementation(async id => ({ tribe: dummyTribes.find(tribe => tribe._id === id) }));
+api.leave.mockImplementation(async id => ({ tribe: dummyTribes.find(tribe => tribe._id === id) }));
+
+
+describe('TribesPage', () => {
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('not signed in', () => {
+
+    it('fetch tribes from api and show tribes on page', async () => {
+      // first no api should be called
+      expect(api.read).toHaveBeenCalledTimes(0);
+
+      const { getAllByRole } = render(<TribesPage onMembershipUpdated={() => {}} />);
+
+      await wait();
+
+      // the tribes from api should be displayed
+      const tribes = getAllByRole('listitem');
+      expect(tribes).toHaveLength(dummyTribes.length + 1); // there is additionally the Suggest Tribe
+
+      tribes.slice(0, -1).forEach((tribe, i) => {
+        expect(tribe).toHaveTextContent(dummyTribes[i].label);
+      });
+
+      // during the test the api should be called only once
+      expect(api.read).toHaveBeenCalledTimes(1);
+    });
+
+    it('the join button should be a link to tribe page', async () => {
+      const { getAllByRole } = render(<TribesPage onMembershipUpdated={() => {}} />);
+
+      await wait();
+
+      // join button
+      const tribes = getAllByRole('listitem').slice(0, -1);
+
+      tribes.forEach((tribe, i) => {
+        const button = rtl.getByText(tribe, 'Join');
+        expect(button).toHaveAttribute('href', `/signup?tribe=${dummyTribes[i].slug}`);
+      });
+    });
+
+  });
+
+  describe('signed in', () => {
+
+    it('show tribes on page', async () => {
+      // first no api should be called
+      expect(api.read).toHaveBeenCalledTimes(0);
+
+      const { getAllByRole } = render(<TribesPage user={dummyUser} onMembershipUpdated={() => {}} />);
+
+      await wait();
+
+      // the tribes from api should be displayed
+      const tribes = getAllByRole('listitem');
+      expect(tribes).toHaveLength(dummyTribes.length + 1); // there is additionally the Suggest Tribe
+
+      tribes.slice(0, -1).forEach((tribe, i) => {
+        expect(tribe).toHaveTextContent(dummyTribes[i].label);
+      });
+
+      // during the test the api should be called only once
+      expect(api.read).toHaveBeenCalledTimes(1);
+    });
+
+    it('user is member of some tribes and not member of others', async () => {
+      const { getAllByText } = render(<TribesPage user={dummyUser} onMembershipUpdated={() => {}} />);
+
+      await wait();
+
+      // the tribes from api should be displayed
+      const join = getAllByText('Join');
+      const leave = getAllByText('Joined');
+      expect(join).toHaveLength(1);
+      expect(leave).toHaveLength(2);
+    });
+
+    it('click join and send api request', async () => {
+      expect(api.join).toHaveBeenCalledTimes(0);
+      const { getAllByText } = render(<TribesPage user={dummyUser} onMembershipUpdated={() => {}} />);
```

Comment:
You could create a mock function to check if onMembershipUpdated got called using `jest.fn()`, something like:

```js
const onMembershipUpdated = jest.fn();
const { getAllByText } = render(<TribesPage user={dummyUser} onMembershipUpdated={onMembershipUpdated} />);
// ...
expect(onMembershipUpdated).toHaveBeenCalled...
```

Reply:
done in e52079d55e2f93259c5a8115df108969cc2f1069